bin/ocr-transform.sh

```diff
@@ -130,4 +130,8 @@ while [[ "$1" = -* ]]; do
     shift
 done
 
+if [[ -d "$SHAREDIR/venv" ]];then
+    . "$SHAREDIR/venv/bin/activate"
+fi
+
```

Comment:
```suggestion
test -d "$SHAREDIR/venv" && . "$SHAREDIR/venv/bin/activate"

```

Reply:
If we decide to run the script with `set -e` in the future (for which there are good reasons), then your formulation would fail half the time.

Reply:
In that case adding `|| true` to my suggestion would be sufficient to avoid such failures.

Reply:
> In that case adding `|| true` to my suggestion would be sufficient to avoid such failures.

Different outcome again: if `activate` is broken (for whatever reason) and fails, then this would be hidden.

> `test ! -d "$SHAREDIR/venv" || . "$SHAREDIR/venv/bin/activate"` would also work.

Sure, but it looks ugly. 

The rest of the scripts here use explicit conditionals a lot, hence more consistent style.

Reply:
I also prefer semi-readable (one space after punctuation), but did not want to change style.

Compound conditional expressions are more potent than simple ones, so yes, I would also argue in general for sole use of `[[..]]` as it makes reading _and_ writing tests easier.

I did use [shellcheck](https://github.com/koalaman/shellcheck) at times without much hurt.

But more importantly: whatever works – we are not likely to win a beauty contest here anyway!